homeassistant/components/nut/device_action.py

```diff
@@ -69,8 +65,15 @@ def _get_command_name(device_action_name: str) -> str:
     return device_action_name.replace("_", ".")
 
 
-def _get_entry_id_from_device_id(hass: HomeAssistant, device_id: str) -> str | None:
+def _get_runtime_data_from_device_id(
+    hass: HomeAssistant, device_id: str
+) -> NutRuntimeData | None:
     device_registry = dr.async_get(hass)
     if (device := device_registry.async_get(device_id)) is None:
         return None
-    return next(entry for entry in device.config_entries)
+    entry = hass.config_entries.async_get_entry(
+        next(entry_id for entry_id in device.config_entries)
```

Comment:
i just took it over, since it is existing code, but i'm not sure if this is correct - assume you have a device which is linked to multiple config entries, than this would only consider the first config entry 🤔

Reply:
I think it would be a bug if the same device was provided by multiple config entries in the same domain

Reply:
ahh ... was not aware, that we are "domain aware" here ... than it makes sense as it is right now 👍